Update OpenAPI Learn Page (#9663)
ballerina-platform/ballerina-dev-website: swan-lake/integration-tools/openapi-tool.md

| `-s \| --service` | This service name is used to identify the service that needs to be documented as an OpenAPI contract.                                                                                                | Optional           |
| `-o\|--output`    | Location of the generated OpenAPI contract. If this path is not specified, the output is written to the same directory from which the command is run.                                                | Optional           |

### Align OpenAPI contract command options

The `align` subcommand of the Ballerina OpenAPI tool supports the following options.

| Command option            | Description                                                                                                                                                                                                               | Mandatory/Optional |
|---------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------|

> | `-i \| --input`           | The path of the OpenAPI contract file (e.g., `my-api.yaml` or `my-api.json`).                                                                                        | Mandatory          |
> | `-o \| --output`          | The directory location where the aligned OpenAPI contract will be generated. The default location is same as the location from which the `bal openapi align` command is executed.                   | Optional           |
> | `-n \| --name`            | The given name will be used to save the aligned OpenAPI contract. The default name is `aligned_ballerina_openapi`.                                                                                                              | Optional           |
> | `-f \| --format`          | The aligned OpenAPI contract will be saved in the given format. The format can be either JSON or YAML. The default format is same as the input file format.                                                              | Optional           |
> | `-t \| --tags`            | The aligned OpenAPI contract will only have the operations with the given tags.                                                                                                                                         | Optional           |
> | `--operations`            | The aligned OpenAPI contract will only have the given operations.                                                                                                                                                       | Optional           |

Reviewer: ```suggestion
| `-i \| --input`           | The path of the OpenAPI contract file (e.g., `my-api.yaml` or `my-api.json`).                                                                                        | Mandatory          |
| `-o \| --output`          | The directory location where the aligned OpenAPI contract will be generated. The default location is the same as the location from which the `bal openapi align` command is executed.                   | Optional           |
| `-n \| --name`            | The given name will be used to save the aligned OpenAPI contract. The default name is `aligned_ballerina_openapi`.                                                                                                              | Optional           |
| `-f \| --format`          | The aligned OpenAPI contract will be saved in the given format. The format can be either JSON or YAML. The default format is the same as the input file format.                                                              | Optional           |
| `-t \| --tags`            | The aligned OpenAPI contract will only have the operations with the given tags.                                                                                                                                         | Optional           |
| `--operations`            | The aligned OpenAPI contract will only have the given operations.                                                                                                                                                       | Optional           |
```

### Flatten OpenAPI contract command options

> The `flatten` subcommand of the Ballerina OpenAPI tool supports the following options.
> 
> | Command option            | Description                                                                                                                                                                                                               | Mandatory/Optional |
> |---------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------|
> | `-i \| --input`           | The path of the OpenAPI contract file (e.g., `my-api.yaml` or `my-api.json`).                                                                                        | Mandatory          |
> | `-o \| --output`          | The directory location where the flattened OpenAPI contract will be generated. The default location is same as the location from which the `bal openapi flatten` command is executed.                   | Optional           |
> | `-n \| --name`            | The given name will be used to save the flattened OpenAPI contract. The default name is `flattened_openapi`.                                                                                                              | Optional           |
> | `-f \| --format`          | The flattened OpenAPI contract will be saved in the given format. The format can be either JSON or YAML. The default format is same as the input file format.                                                              | Optional           |
> | `-t \| --tags`            | The falttened OpenAPI contract will only have the operations with the given tags.                                                                                                                                         | Optional           |
> | `--operations`            | The falltened OpenAPI contract will only have the given operations.                                                                                                                                                       | Optional           |
> 

Reviewer: ```suggestion
The `flatten` subcommand of the Ballerina OpenAPI tool supports the following options.

| Command option            | Description                                                                                                                                                                                                               | Mandatory/Optional |
|---------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------|
| `-i \| --input`           | The path of the OpenAPI contract file (e.g., `my-api.yaml` or `my-api.json`).                                                                                        | Mandatory          |
| `-o \| --output`          | The directory location where the flattened OpenAPI contract will be generated. The default location is the same as the location from which the `bal openapi flatten` command is executed.                   | Optional           |
| `-n \| --name`            | The given name will be used to save the flattened OpenAPI contract. The default name is `flattened_openapi`.                                                                                                              | Optional           |
| `-f \| --format`          | The flattened OpenAPI contract will be saved in the given format. The format can be either JSON or YAML. The default format is the same as the input file format.                                                              | Optional           |
| `-t \| --tags`            | The flattened OpenAPI contract will only have the operations with the given tags.                                                                                                                                         | Optional           |
| `--operations`            | The flattened OpenAPI contract will only have the given operations.                                                                                                                                                       | Optional           |

```

| `licenseURL: string`        | You can use this to add the URL details regarding the full text of the license.                                                                                                                                                                                                                                                           | Optional          |
| `embed: boolean`             | Turns on generating OpenAPI documentation for the service for [introspection endpoint](https://ballerina.io/spec/http/#236-openapi-specification-resources) support when used with `true` in the annotation.                                                                                                                                                                                               | Optional          |

## OpenAPI Contract Modifier

> ### Modify the OpenAPI contract using Ballerina-Preferred naming conventions
> 
> The `align` subcommand alignes the OpenAPI contract file according to the naming conventions of Ballerina. The Ballerina-specific name extension `x-ballerina-name` is added to the schemas, including query names and object property names, which cannot be modified directly.

Reviewer: ```suggestion
### Modify the OpenAPI contract using Ballerina-Preferred naming conventions

The `align` subcommand aligns the OpenAPI contract file according to Ballerina's naming conventions. The Ballerina-specific name extension `x-ballerina-name` is added to the schemas, including query and object property names, which cannot be modified directly.
```

```
bal openapi align [-i | --input] <openapi-contract-file-path>
                  [-o | --output] <output-file-path>
                  [-n | --name] <generated-file-name>
                  [-f | --format] [json|yaml]
                  [-t | --tags] <tag-names>
                  [--operations] <operation-names>
```

> The command-line options are as follows.
> 
> | Command option            | Description                                                                                                                                                                                                               | Mandatory/Optional |
> |---------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------|
> | `-i \| --input`           | The path of the OpenAPI contract file (e.g., `my-api.yaml` or `my-api.json`).                                                                                        | Mandatory          |
> | `-o \| --output`          | The directory location where the aligned OpenAPI contract will be generated. The default location is same as the location from which the `bal openapi align` command is executed.                   | Optional           |
> | `-n \| --name`            | The given name will be used to save the aligned OpenAPI contract. The default name is `aligned_ballerina_openapi`.                                                                                                              | Optional           |
> | `-f \| --format`          | The aligned OpenAPI contract will be saved in the given format. The format can be either JSON or YAML. The default format is same as the input file format.                                                              | Optional           |
> | `-t \| --tags`            | The aligned OpenAPI contract will only have the operations with the given tags.                                                                                                                                         | Optional           |
> | `--operations`            | The aligned OpenAPI contract will only have the given operations.                                                                                                                                                       | Optional           |

Reviewer: ```suggestion
The command-line options are as follows.

| Command option            | Description                                                                                                                                                                                                               | Mandatory/Optional |
|---------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------|
| `-i \| --input`           | The path of the OpenAPI contract file (e.g., `my-api.yaml` or `my-api.json`).                                                                                        | Mandatory          |
| `-o \| --output`          | The directory location where the aligned OpenAPI contract will be generated. The default location is the same as the location from which the `bal openapi align` command is executed.                   | Optional           |
| `-n \| --name`            | The given name will be used to save the aligned OpenAPI contract. The default name is `aligned_ballerina_openapi`.                                                                                                              | Optional           |
| `-f \| --format`          | The aligned OpenAPI contract will be saved in the given format. The format can be either JSON or YAML. The default format is the same as the input file format.                                                              | Optional           |
| `-t \| --tags`            | The aligned OpenAPI contract will only have the operations with the given tags.                                                                                                                                         | Optional           |
| `--operations`            | The aligned OpenAPI contract will only have the given operations.                                                                                                                                                       | Optional           |
```

components:
  schemas:
    album:
      required:
        - _id
        - artist
        - title
      type: object
      properties:
        _id:
          type: string
        title:
          type: string
        artist:
          type: string
      additionalProperties: false
```

**Output OpenAPI contract file:**

```yaml
paths:
  /albums:
    get:
      tags:
        - albums
      operationId: getAlbums
      parameters:
        - name: _artists_
          in: query
          schema:
            type: array
            items:
              type: string
          x-ballerina-name: artists   -->//Ballerina name extension
      responses:
        "200":
          description: Ok
          content:
            application/json:
              schema:
                type: array
                items:
                  $ref: "#/components/schemas/Album"
components:
  schemas:
    Album:
      required:
        - _id
        - artist
        - title
      type: object
      properties:
        _id:
          type: string
          x-ballerina-name: id
        artist:
          type: string
        title:
          type: string
      additionalProperties: false
```

> ### Modify the OpenAPI Contract by defining named objects for inline objects
> 
> The `flatten` subcommand makes the OpenAPI contract more readable by relocating all inline embedded schemas to the components section.
> 
> ```
> bal openapi flatten [-i | --input] <openapi-contract-file-path>
>                     [-o | --output] <output-file-path>
>                     [-n | --name] <generated-file-name>
>                     [-f | --format] [json|yaml]
>                     [-t | --tags] <tag-names>
>                     [--operations] <operation-names>
> ```
> 
> The command-line options are as follows.
> 
> | Command option            | Description                                                                                                                                                                                                               | Mandatory/Optional |
> |---------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------|
> | `-i \| --input`           | The path of the OpenAPI contract file (e.g., `my-api.yaml` or `my-api.json`).                                                                                        | Mandatory          |
> | `-o \| --output`          | The directory location where the flattened OpenAPI contract will be generated. The default location is same as the location from which the `bal openapi flatten ` command is executed.                   | Optional           |
> | `-n \| --name`            | The given name will be used to save the flattened OpenAPI contract. The default name is `flattened_openapi`.                                                                                                              | Optional           |
> | `-f \| --format`          | The flattened OpenAPI contract will be saved in the given format. The format can be either JSON or YAML. The default format is same as the input file format.                                                              | Optional           |
> | `-t \| --tags`            | The falttened OpenAPI contract will only have the operations with the given tags.                                                                                                                                         | Optional           |
> | `--operations`            | The falltened OpenAPI contract will only have the given operations.                                                                                                                                                       | Optional           |

Reviewer: ```suggestion
### Modify the OpenAPI Contract by defining named objects for inline objects

The `flatten` subcommand makes the OpenAPI contract more readable by relocating all inline embedded schemas to the components section.

```
bal openapi flatten [-i | --input] <openapi-contract-file-path>
                    [-o | --output] <output-file-path>
                    [-n | --name] <generated-file-name>
                    [-f | --format] [json|yaml]
                    [-t | --tags] <tag-names>
                    [--operations] <operation-names>
```

The command-line options are as follows.

| Command option            | Description                                                                                                                                                                                                               | Mandatory/Optional |
|---------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------|
| `-i \| --input`           | The path of the OpenAPI contract file (e.g., `my-api.yaml` or `my-api.json`).                                                                                        | Mandatory          |
| `-o \| --output`          | The directory location where the flattened OpenAPI contract will be generated. The default location is the same as the location from which the `bal openapi flatten ` command is executed.                   | Optional           |
| `-n \| --name`            | The given name will be used to save the flattened OpenAPI contract. The default name is `flattened_openapi`.                                                                                                              | Optional           |
| `-f \| --format`          | The flattened OpenAPI contract will be saved in the given format. The format can be either JSON or YAML. The default format is the same as the input file format.                                                              | Optional           |
| `-t \| --tags`            | The flattened OpenAPI contract will only have the operations with the given tags.                                                                                                                                         | Optional           |
| `--operations`            | The flattened OpenAPI contract will only have the given operations.                                                                                                                                                       | Optional           |
```

Author: Fixed with https://github.com/ballerina-platform/ballerina-dev-website/pull/9663/commits/0c05fb31b68d6069de8d405d5d09cdbf11ba43e8
